pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,464 @@
+            show_progress_message();
+
+            var deploy_btn = $('#deploy-app-start');
+            deploy_btn.prop("disabled", true);
+
+            if (isJsonString(jsonData)) {
+                var jdata = JSON.parse(jsonData);
+                if(jdata.kind === "List") {
+                    if (jdata.items) {
+                        for (var i = 0; i < jdata.items.length; i++) {
+                            var ent_json = jdata.items[i];
+                            if (ent_json.kind === SERVICE) {
+                                services.push(ent_json);
+                            } else if (ent_json.kind === POD) {
+                                pods.push(ent_json);
+                            } else if (ent_json.kind === RC) {
+                                rcs.push(ent_json);
+                            } else if (ent_json.kind === NS) {
+                                namespaces.push(ent_json);
+                            }
+                        }
+                    } else {
+                        text = _("Unable to read the Kubernetes application manifest file. ");
+                        file_note.show().text(text);
+                        return;
+                    }
+                } else {
+                    if (jdata.kind === SERVICE) {
+                        services.push(jdata);
+                    } else if (jdata.kind === POD) {
+                        pods.push(jdata);
+                    } else if (jdata.kind === RC) {
+                        rcs.push(jdata);
+                    } else if (jdata.kind === NS) {
+                        namespaces.push(jdata);
+                    } else {
+                        text = _("Unsupported Entity. ");
+                        file_note.show().text(text);
+                        return;
+                    }
+                }
+            } else {
+                text = _("Unable to read the Kubernetes application manifest file. ");
+                file_note.show().text(text);
+                return;
+            }
+
+            create_everything(namespace, services, rcs, pods)
+                .progress(function(code, response, n, total) {
+                    //TODO Progress bar
+
+                })
+                .done(function() {
+                    /* code gets run when everything is created */
+                    disable_deploy_button();
+                    hide_progress_message();
+                     $('#deploy-app-dialog').modal('hide');
+
+                })
+                .fail(function(ex, response) {
+                    /* ex containst the failure */
+                    enable_deploy_button();
+                    var jdata = JSON.parse(response);
+                    var err_msg = jdata.message;
+                    if(jdata.code === 409){
+                        err_msg = "Please create another namespace for "+ jdata.details.kind + "\""+ jdata.details.id +"\"";
```

Comment:
We should use cockpit.format() here, and then wrap the text in _("") so it can be translated.